luminance-gl/src/gl33/state.rs

```diff
@@ -35,6 +35,39 @@ impl BindingStack {
   }
 }
 
+struct CachedValue<T>
+where
+  T: PartialEq,
+{
+  value: Option<T>,
+}
```

Comment:
`Cached` should be enough, and you can simply make it a wrapper type:

```suggestion
struct Cached<T>(Option<T>) where T: PartialEq;
```